<launch>
  
  <arg name="serial_no_cameraL"    			default="838212073567"/> 			<!-- Note: Replace with actual serial number -->
  
  <arg name="serial_no_cameraR"    			default="838212073332"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="cameraL"              			default="camL"/>		  
  <arg name="cameraR"              			default="camR"/>		

  <arg name="tf_prefix_cameraL"         default="$(arg cameraL)"/>
  <arg name="tf_prefix_cameraR"         default="$(arg cameraR)"/>
  
  <arg name="initial_reset"             default="false"/>
  <arg name="filters"             default="pointcloud"/>
  

<!-- launch the cameras -->


<group ns="$(arg cameraR)"> 
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">

      
      <arg name="serial_no"             value="$(arg serial_no_cameraR)"/>
      <arg name="tf_prefix"         		value="$(arg tf_prefix_cameraR)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
      <arg name="filters"		value="$(arg filters)"/>
    </include>
 </group> 


<include file="$(find cob_openni2_tracker)/launch/body_tracker_nodelet.launch">
      
      <arg name="standalone_without_camera_driver"   value="true"/>
	<arg name="camera_name"   value="$(arg cameraL)"/>
	
</include>
 


<!-- launch the tf publishers -->

<node name="tf_base_link" pkg="tf" type="static_transform_publisher" output="screen" args="0 0 0 0 0 0 world base_link 50"/>

<node name="tf_camL_link" pkg="tf" type="static_transform_publisher" output="screen" args="-0.8 0.265 0.2 -1.5708 0 -1.5708 world camL_link 500" />

<node name="tf_camR_link" pkg="tf" type="static_transform_publisher" output="screen" args="-0.8 0.685 0.15625 0 0 0 world camR_link 500" />

<!-- launch the robot in gazebo -->

<include file="$(find ur5e_egp50_moveit_config)/launch/gazebo.launch" />





<!-- launch Moveit -->

<include file="$(find ur5e_egp50_moveit_config)/launch/ur5e_egp50_moveit_planning_execution.launch">
	<arg name="sim" value="true"/>
</include>


<!-- launch rviz -->

<node name="rviz" pkg="rviz" type="rviz" args="-d $(find seher)/launch/rviz_config_home.rviz"> 
	<rosparam command="load" file="$(find ur5e_egp50_moveit_config)/config/kinematics.yaml"/>
</node>

<!-- launch the pointcloud computing -->

<node name="pointcloud_computing" pkg="seher" type="pointcloud_computing_home"/>



<!-- launch the distance calculation -->

<node name="distance_calculation" pkg="seher" type="distance_calculation"/>


</launch>
